samples/snippets/revoke_dataset_access.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def revoke_dataset_access(your_dataset_id, your_entity_id):
+
+    # [START bigquery_revoke_dataset_access]
+    from google.cloud import bigquery
+
+    # Construct a BigQuery client object.
+    client = bigquery.Client()
+
+    original_your_dataset_id = your_dataset_id
+    original_your_entity_id = your_entity_id
+    # [START bigquery_revoke_dataset_access_read_session]
+    your_dataset_id = "dataset-for-read-session"
+    your_entity_id = "entity-for-read-session"
+    # [END bigquery_revoke_dataset_access_read_session]
+    your_dataset_id = original_your_dataset_id
+    your_entity_id = original_your_entity_id
+
+    dataset_id = your_dataset_id
+    entity_id = your_entity_id
```

Comment:
- Anything between `START` and `END` (region tags) will get included in the docs. If two `START`s and `END`s with the same region tag are found in the same file, those code snippets are concatenated together. The nested region tags you've written can sometimes be used in a tutorial where you want to explain a smaller section of the code, but I don't think that's relevant here.
- I find the inclusion of `your_dataset_id` and `dataset_id` confusing. It doesn't appear to be necessary to have that extra layer of indirection.
- When you provide example values, please make sure they are valid / look like the same pattern as would be expected by the API. For example `-` is not an allowed character in a dataset ID. `"your_dataset_id"` would be a more appropriate string. Likewise for entity ID. I recommend playing with the API (the "Try this API" explorer in the REST documentation can be useful for this: https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets/get?apix_params=%7B%22projectId%22%3A%22swast-scratch%22%2C%22datasetId%22%3A%22my_dataset%22%7D) to see what some example IDs look like and use them as a pattern.
- [Less-informed opinion] Since we know people don't read all that closely, it's good to put all the things the developer has to modify first in the code sample.


```suggestion
def revoke_dataset_access(dataset_id, entity_id):
    original_dataset_id = dataset_id
    original_entity_id = entity_id
    # [START bigquery_revoke_dataset_access]
    # TODO(developer): Set the dataset ID to the dataset where you are revoking access.
    dataset_id = "your_dataset_id"

    # TODO(developer): Set the entity ID of the email or group from whom you are revoking access.
    entity_id = "user-or-group-to-remove@example.com"

    # [END bigquery_revoke_dataset_access]
    dataset_id = original_dataset_id
    entity_id = original_entity_id

    # [START bigquery_revoke_dataset_access]
    from google.cloud import bigquery

    # Construct a BigQuery client object.
    client = bigquery.Client()

```